File: 119. Pascal's Triangle II.cpp
class Solution {
public:
    vector<int> getRow(int rowIndex) {
        if (rowIndex == 0) return {1};
        if (rowIndex == 1) return {1,1};
        vector<int> pre {1,1};
        for (int i = 2; i <= rowIndex; i++)
        {
            vector <int> temp (1,1);
            for (int j = 1; j < pre.size(); j++)
                temp.push_back(pre[j - 1] + pre[j]);
            temp.push_back(1);
            pre = temp;
        }
        return pre;
    }
};

File: 119. Pascal's Triangle II.py
class Solution(object):
    def getRow(self, rowIndex):
        """
        :type rowIndex: int
        :rtype: List[int]
        """
        if rowIndex == 0:
            return [1]
        if rowIndex == 1:
            return [1, 1]
        pre = [1, 1]
        for i in xrange(2, rowIndex + 1):
            temp = [1]
            for j in xrange(1, len(pre)):
                temp += [pre[j-1] + pre[j]]
            temp += [1]
            pre = temp[:]
        return pre
